File: app/src/main/java/com/dkaluta/prosary/models/EternalRestPlacement.kt
package com.dkaluta.prosary.models

/** Where (if at all) "Eternal rest grant unto them, O Lord" is prayed for the faithful departed. */
enum class EternalRestPlacement {
    None,

    /** Pray it after the Glory Be of every decade. */
    AfterEachDecade,

    /** Pray it once, near the end, before the closing prayers. */
    AtEndOnly;

    val displayName: String
        get() = when (this) {
            None -> "Don't Include"
            AfterEachDecade -> "After Each Decade"
            AtEndOnly -> "Once, Near the End"
        }
}

File: iOS/Prosary/Models/EternalRestPlacement.swift
//
//  EternalRestPlacement.swift
//  Prosary
//

import Foundation

/// Where (if at all) "Eternal rest grant unto them, O Lord" is prayed for the faithful departed.
enum EternalRestPlacement: String, Codable, CaseIterable, Identifiable {
  case none
  /// Pray it after the Glory Be of every decade.
  case afterEachDecade
  /// Pray it once, near the end, before the closing prayers.
  case atEndOnly

  var id: String { rawValue }

  var displayName: String {
    switch self {
    case .none:           return String(localized: "eternalRest.none", defaultValue: "Don't Include")
    case .afterEachDecade: return String(localized: "eternalRest.afterEachDecade", defaultValue: "After Each Decade")
    case .atEndOnly:      return String(localized: "eternalRest.atEndOnly", defaultValue: "Once, Near the End")
    }
  }
}
